enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,492 @@
+)
+
+type DeviceDiscoveryPolicyType string
+
+const (
+	// The DeviceDiscoveryPolicies that will be supported by the LSO.
+	// These Discovery policies will based on lsblk's type output
+	// Disk represents a device-type of disk
+	Disk DeviceDiscoveryPolicyType = "disk"
+	// Part represents a device-type of partion
+	Part DeviceDiscoveryPolicyType = "part"
+)
+
+type DeviceInclusionSpec struct {
+	// DeviceTypes that should be used for automatic detection. This would be one of the types supported
+	// by the local-storage operator. Currently, the supported types are: disk, part
+	// If the list is empty the default value will be `[disk]`.
+	DeviceTypes []DeviceDiscoveryPolicyType `json:"deviceType"`
+
+	// DeviceMechanicalProperty denotes whether Rotational or NonRotational disks should be used.
+	// by default, it selects both
+	// +optional
+	DeviceMechanicalProperty DeviceMechanicalProperty `json:"deviceMechanicalProperty"`
+
+	// MinSize is the minimum size of the device which needs to be included
+	// +optional
+	MinSize resource.Quantity `json:"minSize"`
+
+	// MaxSize is the maximum size of the device which needs to be included
+	// +optional
+	MaxSize resource.Quantity `json:"maxSize"`
+
+	// Models is a list of device models. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Models []string `json:"models"`
+
+	// Vendors is a list of device vendors. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Vendors []string `json:"vendors"`
+}
+
+type AutoDetectVolumePhase string
+
+const (
+	DiscoveringPhase AutoDetectVolumePhase = "Discovering"
+	FailedPhase      AutoDetectVolumePhase = "Failed"
+	DiscoveredPhase  AutoDetectVolumePhase = "Discovered"
+	ProvisionedPhase AutoDetectVolumePhase = "Discovered"
+)
+
+type AutoDetectVolumeStatus struct {
+	Phase                       AutoDetectVolumePhase `json:"phase,omitempty"`
+	TotalDiscoveredDeviceCount  *int32                `json:"totaldiscovereddevicecount,omitempty"`
+	TotalProvisionedDeviceCount *int32                `json:"totalprovisioneddevicecount,omitempty"`
+	TimeStramp                  `json:"phase,omitempty"`
+}
+```
+### Workflow of LocalPV creation via AutoDetectVolume CR
+- Once the `AutoDetectVolume` is created, the controller will:
+  - configure the local-static-provisioner to make a new StorageClass based on certain directories on the selected nodes.
+  - assign diskmaker daemons to the selected nodes.
+- The diskmaker daemon will find devices that match the disovery policy and symlink them into the directory that the local-static-provisioner is watching.
+- Possible `status.phase` values for `AutoDetectVolume`
+  - `Discovering`
+  - `Failed`
+  - `Discovered`
+  - `Provisioned`
+  
+Example of `AutoDetectVolume` CR:
+
+```yaml
+apiVersion: local.storage.openshift.io/v1alpha1
+kind: AutoDetectVolume
```

Comment:
Instead of `AutoDetectVolume`, what about naming it `LocalVolumeBatch` or `LocalVolumeGroup`?

Reply:
I will vote for `LocalVolumeGroup`.